enhancements/update/versioning-cincinnati-api.md

```diff
@@ -0,0 +1,122 @@
+
+
+## Summary
+As of today, we declare the schema version in Cincinnati Graph URI. We currently, do not embed the version information
+in json schema. The enhancement will embed the version information in the json schema. This has the benefit of being
+easy to implement and to pass around if we ship Cincinnati JSON over non-HTTP channels. The enhancement will also begin
+versioning Cincinnati payloads by media type, for example `application/vnd.redhat.cincinnati.v1+json`.
+The move to supplying requested version in media type field will require us to move towards a generic un-versioned Graph
+URI, with the move, we'll still support old URI for foreseeable future.
+
+## Motivation
+
+### Goals
+1. Embed version information in json schema.
+2. Implement media type versioning for clients requesting a specific schema version of Cincinnati
+3. Move to a generic un-versioned URI for Cincinnati
+4. Cincinnati continues to support older clients after adding support for new clients for foreseeable future.
+5. Add metrics to identify hits on URIs and versions.
+
+### Non-Goals
+
+## Proposal
+
+### Implementation
+
+#### Version information in json schema
+Embed the version information in json schema. The updated json schema should look like:
+```json
+{
+    "version": 1,
+    "nodes": [],
+    "edges": []
+}
+```
+
+#### Media Type Versioning
+Currently, the versioning is done by using the versioned URI and does not support media type versioning. The user should
+be able to request a specific version of cincinnati graph by supplying the version in media type field of the request.
+for example `application/vnd.redhat.cincinnati.v1+json`. Here are some of the
+[examples of media type versioning](https://github.com/opencontainers/image-spec/blob/v1.0.1/media-types.md).  
+In the event where the media type is not provided, or `application/json` is provided, cincinnati should return a default
+response that is compatible with the oldest supported version so as not to break old clusters from upgrading.
+
+
+#### URI Change
+The current URI follows `{URL}/v1/graph`. The new un-versioned URI will follow `{URL}/graph`. The existing URI can be
+redirected to the new URI. This can be done through cincinnati. In the case where an old URI is being called, cincinnati
+will by default return the graph that is compatible with the oldest supported version.
+
+
+#### Version metrics
+Cincinnati should also implement a way to understand how many clients are consuming the various versions.
+This can be done by implementing metrics for old URI vs new URI in Cincinnati. Similarly, metrics should be implemented
+to understand the consumption of various media types (versions). This will help us to make decisions on deprecating the
+old versions and URI. For example, a single `cincinnati_pe_http_response_total` counter with labels for `uri` and `version`
+
+
+### Risks and Mitigations
```

Comment:
The HTTP spec lists some disadvantages in [its proactive negotiation section][1].  We should reference the HTTP spec and bring those over, and explain why we think they are acceptable in this case.  For example:

```Markdown
Another disadvantage listed in the HTTP RFC is inefficiency when  only a small percentage of responses have multiple representations.
But in this case, including the desired/acceptable version(s) in a per-request header is about as efficient as hitting a versioned URI that the service supports, and much more efficient than [the alternatives](#uri-versioning) for supporting versioned URIs when the service does not support the client's preferred version.
```

Where `#uri-versioning` links [the relevant alternatives subsection][2].

[1]: https://datatracker.ietf.org/doc/html/rfc7231#section-3.4.1
[2]: https://github.com/openshift/enhancements/pull/870#discussion_r694366627

Reply:
added